src/utils/deepObject.js

```diff
@@ -0,0 +1,91 @@
+const _ = require('lodash');
+const logger = require('../logger');
+
+const validateKeys = (newKeys) => newKeys.reduce((validatedKeys, key) => {
+	if (key && key !== '') {	// remove invalid pathes
+		validatedKeys.push(key);
+	}
+	return validatedKeys;
+}, []);
+
+/**
+ * @param  {...String||Array} paths 
+ */
+const pathsToArray = (...paths) => {
+	try {
+		let keys = [];
+		(paths || []).forEach((path) => {
+			if (path) {	// avoid null and undefined
+				let newKeys = [];
+				if (typeof path === 'string') {
+					// for dot notation y.asd.asdasd.sdada and array notations like a[0]
+					newKeys = path.replace(/\[/g, '.').replace(/]/g, '.').split('.');
+				} else if (Array.isArray(path)) {
+					newKeys = pathsToArray(...path);
+				} else {
+					throw new Error('Wrong type is adding as path in deepObjectProps.');
+				}
+
+				keys = [...keys, ...validateKeys(newKeys)];
+			}
+		});
+		return keys;
+	} catch (err) {
+		logger.error(err);
+		return [];
+	}
+};
+
+/**
+ * Returns the value of a (nested) attribute
+ * @param {*} item a value
+ * @param {String} pathString a string describing the attribute to select
+ * @example
+ * getSimple({foo: 'bar'}, 'foo') => 'bar'
+ * getSimple({foo: {bar: 'baz'}}, 'foo.bar') => 'baz'
+ */
+const getSimple = (item, pathString) => {
+	if (!item || !pathString) return item;
+	try {
+		let chain = pathString.split('.');
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['chain' is never reassigned. Use 'const' instead. (prefer-const)](https://app.codacy.com/app/schul-cloud/schulcloud-server/pullRequest?prid=4646529)